### ISSUE TITLE
Editing custom color: keyboard trap and controls not operable with the keyboard

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51234]
### Description

In the color palette, custom colors can't be edited with the keyboard. The `PaletteEdit` component doesn't seem to have been designed with keyboard accessibility in mind in the first place.

## Keyboard trap
The color picker of the custom colors doesn't close when pressing the Escape key. Users are trapped within the color picker. The only workaround I could fidn to exti this keyboard trap is by using the 'navigate region' keyboard shortcut Ctrl + backtick.

## Clickable div elements
The color name can't be edited with the keyboard. [It's a clickable div element](https://github.com/WordPress/gutenberg/blob/8ee14b0943e445ea017c179ff20934e24fb16905/packages/components/src/palette-edit/index.tsx#L195-L198). When clicked with the moust, it switches to an editable input field. Given it's a clickable, non-focusable, non =-natively interactive element, it can'e be operated with the keyboard.

I'd like to note that this is not the first time clickable div elements have been merged in the codebase, see for example https://github.com/WordPress/gutenberg/issues/42461 and https://github.com/WordPress/gutenberg/pull/42668. As a follow-up, a new issue to [Improve accessibility linting of styled/polymorphic components](https://github.com/WordPress/gutenberg/issues/42540) was created but I'm not sure there has been much progress there.

Using clickable div elements reveals a couple facts that worry me the most:
- Lack of knowledge of the _most basic_ accessibility principles and best practices by contributors and reviewsers. 
- Total lack of manual keyboard testing. It takes just two minutes of manual testing to realize this pieace of the UI can't be used with the keyboard.

While some form of linting could certainly help, seems to me there's a serious problem of lack of knowledge here. The relevant code tells us a bit more:

```
return (
    <PaletteItem
        className={ isEditing ? 'is-selected' : undefined }
        as="div"
        onClick={ onStartEditing }
```

- A view component is intentionally rendered as a div element.
- An onclick event handler is attached to the div element.

The problem with this code is so self-evident that I'm not sure it something that was just overlooked or rushed out. It's a serious lack of awareness and knowledge that would warrant some kind of discussion at a higher level. I see it as something more related to the process rather than individuals. I'm not sure this kind of things help with meeting the quality level we'd like to see in this project. 

## Additional issues

1
The 'Remove color' button can't be used witht he keyboard. It's rendered only when the color picker is open. As such, it's impossible to reach it with the keyboard. It's impossible to remove colors when using the keyboard.

2
When editing the custom colors, the rounded color button is unlabeled. Also, seems to me it's just a button that doesn't do anything. Apparently, there's no onClick prop. The actual click happens on the div element. Technically, this button is a `it's styled( CircularOptionPicker.Option )`. This component should be improved to make sure labeling is required.

3
the list of custom colors is a `ItemGroup` component. By default, it comes with a role of `list` which is equivalent to a `<ul>` element. As such, its children are expected to be list items. According to [the component documentation](https://github.com/WordPress/gutenberg/blob/53a648baa9e2c94b6d0f02636998948e24f9f106/packages/components/src/item-group/item-group/README.md#L11) it `should be used in combination with the Item sub-component`.  Instead, its children are div elements, which makes using `ItemGroup` pointless in the first place. 
I'm not sure this is ideal, as it's pretty evident this component has been used in this case without even reading its documentation. Also, `ItemGroup` doesn't enforce any check to make sure it is used correctly thus it's open to abuse. 

In an ideal world, education and knowledge should make contributors use components responsibly, according to the components intended usage. However, it's a fact that sometimes components appear to be used in some incorrect way. I'd tend to think components should be improved to enforce their correct usage.

4
Noting that some of the issues reported here are caught by the a11y linting and checks [implemented in the Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-paletteedit--default). But that's only in the Storybook.

Regarding the last points, seems to me it would be beneficial to have some higher level conversation with the `components` package maintainers / contributors to see how we can improve both the components themselves and their actual usage.

### Step-by-step reproduction instructions

Preparatory steps:
- Use the mouse / trackpad tor a faster set-up.
- Go to Site Editor > Styles > Colors > Palette > Custom 
- Add a couple custom colors and click 'Done'.
- Click the 'Color options' ellipsis button then click 'Show details'.
- At this point the UI shows the `PaletteEditListView` component.

<img width="303" alt="Screenshot 2023-06-01 at 14 29 39" src="https://github.com/WordPress/gutenberg/assets/1682452/d600c4f7-ab10-4d1d-ba6c-bdca231ac765">

- Use the keyboard.
- Focus one of the custom colors and press Enter.
- The color picker opens. Tabbing is constrained within the color picker. This is the expected behavior.
- Press Escape to close the color picker.
- Nothing happens.
- Expected: the color picker to close and focus to be moved back to the color button that opened the color picker.
- Try to edit the color name by only using the keyboard.
- Observe that it's impossible to move focus to the color name and edit it.
- Try to remove a single color by only using the keyboard.
- Observe the 'Remove color` button only renders when the color picker is open. As such, it can't be reached with the keyboard.
- Observe the color button on the left of the color name is unlabeled.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
